client/tests/integration/smartcontracts/create_nft_for_every_user_trigger/src/lib.rs

```diff
@@ -9,13 +8,15 @@ extern crate panic_halt;
 use alloc::{format, string::ToString};
 
 use iroha_trigger::prelude::*;
+use lol_alloc::{FreeListAllocator, LockedAllocator};
+
+#[global_allocator]
+static ALLOC: LockedAllocator<FreeListAllocator> = LockedAllocator::new(FreeListAllocator::new());
```

Comment:
While I appreciate the ~5 times less memory overhead, I am a bit uncertain if it's as well tested as the default allocator, and that's what the author warns about as well. No big objection about introducing it, though.

Reply:
Seems to work better than now-unmaintained and leaky `wee_alloc` at least..

Reply:
> While I appreciate the ~5 times less memory overhead, I am a bit uncertain if it's as well tested as the default allocator, and that's what the author warns about as well. No big objection about introducing it, though.

I didn't introduce it in this PR, it was introduced by @Erigara to mitigate memory leak in `wee_alloc`. Having said that, I agree with you, and after this PR is merged users will be given freedom to choose allocator, i.e. they will no longer be pinned down to allocator chosen by us